libs/deepagents-cli/deepagents_cli/skill_loader.py

```diff
@@ -0,0 +1,186 @@
+"""Skill loader for parsing and loading agent skills from SKILL.md files.
+
+This module implements Anthropic's agent skills pattern with YAML frontmatter parsing.
+Each skill is a directory containing a SKILL.md file with:
+- YAML frontmatter (name, description required)
+- Markdown instructions for the agent
+- Optional supporting files (scripts, configs, etc.)
+
+Example SKILL.md structure:
+```markdown
+---
+name: web-research
+description: Structured approach to conducting thorough web research
+---
+
+# Web Research Skill
+
+## When to Use
+- User asks you to research a topic
+...
+```
+"""
+
+import re
+from pathlib import Path
+from typing import TypedDict
+
+
+class SkillMetadata(TypedDict):
+    """Metadata for a skill."""
+
+    name: str
+    """Name of the skill."""
+
+    description: str
+    """Description of what the skill does."""
+
+    path: str
+    """Path to the SKILL.md file."""
+
+
+class SkillLoader:
+    """Loader for agent skills with YAML frontmatter parsing.
+
+    Skills are organized as:
+    skills/
+    ├── skill-name/
+    │   ├── SKILL.md        # Required: instructions with YAML frontmatter
+    │   ├── script.py       # Optional: supporting files
+    │   └── config.json     # Optional: supporting files
+
+    Example:
+        ```python
+        loader = SkillLoader(skills_dir="~/.deepagents/skills")
+        skills = loader.load_skills()
+        for skill in skills:
+            print(f"{skill['name']}: {skill['description']}")
+        ```
+    """
+
+    def __init__(self, skills_dir: str | Path = "~/.deepagents/skills") -> None:
```

Comment:
Let's avoid hard-coding a defaul there, and if possible restrict input to just Path. This isn't user facing, so wide inputs are bad